### ISSUE TITLE
Nowe metody dostawy -   Odbiór osobisty w punkcie sprzedawcy

### ISSUE THREAD

**Issue by PawelTaberski** [allegro/allegro-api#1079]
### REST API
Udostępniliśmy nowe metody dostawy. Identyfikatory nowych metod możesz uzyskać zasobem [GET/sale/delivery-methods](https://developer.allegro.pl/documentation/#/offer-management/get_sale_delivery_methods):

- Odbiór osobisty w punkcie sprzedawcy - id **afa68f62-0154-4556-9ee4-3f97c3c5bdca**
- Odbiór osobisty w punkcie sprzedawcy pobranie - id **b6d6e4cf-67b3-4a4f-a38d-724335e9734f**

**Pamiętaj!** Punkty odbioru osobistego dodasz dedykowanym zasobem /points-of-service, więcej informacji na ten temat znajdziesz w [naszym poradniku](https://developer.allegro.pl/news/2017-08-11-punkty_odbioru/).

### WebAPI
Identyfikatory nowych metod możesz uzyskać zasobem [doGetShipmentData](https://allegro.pl/webapi/documentation.php/show/id,624), bądź [doGetSellFormFieldsForCategory](https://allegro.pl/webapi/documentation.php/show/id,782):

- Odbiór osobisty w punkcie sprzedawcy - fidy: 
   - 404 (pierwsza sztuka), 
   - 504 (kolejna sztuka), 
   - 604 (ilość w paczce)
- Odbiór osobisty w punkcie sprzedawcy pobranie - fidy: 
   - 405 (pierwsza sztuka), 
   - 505 (kolejna sztuka), 
   - 605 (ilość w paczce)

**Comment by mack3d** [comment]
Czym te dostawy się różnią od: "Odbiór osobisty" ?

**Comment by PawelTaberski** [comment]
Tak, dzięki tej nowej metodzie będziemy w stanie m.in. wyświetlić na mapie, którą klient widzi w procesie zakupowym punkty odbioru osobistego. Dlatego zachęcam do odbioru osobistego na nową opcję dostawy. Docelowo będzie tylko "Odbiór osobisty w punkcie sprzedawcy". Dzięki zmianie wybrany punkt pojawi się także w mailu, zwrócimy go przez API.

**Comment by PhotoSoft** [comment]
Dlaczego WebAPI zwraca takie długie nazwy?

> 404	**Odbiór w punkcie po przedpłacie -** Odbiór osobisty w punkcie sprzedawcy (pierwsza sztuka)
> 405	**Odbiór w punkcie -** Odbiór osobisty w punkcie sprzedawcy pobranie (pierwsza sztuka)

To co wytłuściłem jest całkowicie zbędne.

**Comment by PawelTaberski** [comment]
@PhotoSoft Masz racje ten pierwszy człon nazwy jest zbędny, zgłosiłem prośbę o jego usunięcie.

**Comment by Onixarts** [comment]
Czy na sandboxie będzie to udostępnione? O ile wystawić z nowymi dostawami się da, o tyle przy zakupie nie da się wybrać odbioru osobistego jeśli w fid35 nie ma go zaznaczonego.

**Comment by PawelTaberski** [comment]
@Onixarts Jeśli będziesz mieć dodane do oferty metodę dostawy "Odbiór osobisty w punkcie sprzedawcy" i określone punkty "Punkty odbioru" na  fodz-ie klient będzie miał mapkę z wyborem. Funkcja działa także na sandboxie.

**Comment by Onixarts** [comment]
Teraz faktycznie zadziałała. We wcześniej wystawionej aukcji, nawet po dodaniu punktów odbioru nie była ta opcja dostępna. Ale póki co działa ;)

**Comment by PawelTaberski** [comment]
@anih Podeślij proszę przez nasz [formularz kontaktowy](https://allegro.pl/pomoc/kontakt?subjectId=517305f2-5ea1-4b3b-9080-d4cc960ab6a1&kategoria=a0f7ce34-c6f6-43e6-af7d-ad39bacce619) pełnego requesta u responsa w formacie XML, sprawdzimy w czym jest problem. Zaznacz proszę, że zgłoszeniu dotyczy ticketa numer 1079 na GitHub-ie

**Comment by anih** [comment]
Nie mam takiej możliwości, fidy jakie były wysyłane od 23 stycznia do 28 stycznia do około 12 to:
`data = {
            "404": 0,
            "504": 0,
            "604": 100
}`
Do dziś do około 12 taki zestaw działał bezproblemowo tak dla wystawiania nowych jak i aktualizacji aukcji.

**Comment by PawelTaberski** [comment]
Ujednoliciliśmy tą liczbę z tym co możliwe jest również do ustawienia przez Moje Allegro. przedtem mogłeś podać tam liczbę większą od 1, ale ostatecznie i tak wartość przekazana była 1 w fid 604.

**Comment by magik092** [comment]
Mam takie lekko głupie pytanie, jakie wartości podać dla pól 404, 405, 406 w doChangeItemFields? Czego bym nie podał to dostaję ERR_INVALID_VALUE_IN_ATTRIB_FIELD ...
Nie mam już pomysłu co mu podać :/

**Comment by PawelTaberski** [comment]
@magik092 A czy pamiętasz, aby przesyłać wszystkie 3 fid-y dla każdej wysyłki. Czyli w przypadku 404 to dodatkowo 504 i 604?

**Comment by magik092** [comment]
Tak, przesyłałem wszystkie trzy z wartością 0, z wartością 1, różnymi typami (fValueInt, fValueFloat, fValueString), nawet pomijałem pole typu, niestety wciąć to samo :(

**Comment by magik092** [comment]
Ooo, przeszło w poniższej konfiguracji. Nie wiem czy z camelCase wartości coś nie miałem namieszane - w kodzie przesyłałem wcześniej fValueFloat, teraz poprawiłem na fvalueFloat i przeszło dla wartości 0.
Problem rozwiązany ;)

`<?xml version="1.0" encoding="UTF-8"?>
<SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="https://webapi.allegro.pl/service.php">
    <SOAP-ENV:Body>
        <ns1:DoChangeItemFieldsRequest>
            <ns1:sessionId>XXX//00_1</ns1:sessionId>
            <ns1:itemId>7682340162</ns1:itemId>
            <ns1:fieldsToModify>
                <ns1:item>
                    <ns1:fid>404</ns1:fid>
                    <ns1:fvalueFloat>0</ns1:fvalueFloat>
                </ns1:item>
                <ns1:item>
                    <ns1:fid>504</ns1:fid>
                    <ns1:fvalueFloat>0</ns1:fvalueFloat>
                </ns1:item>
                <ns1:item>
                    <ns1:fid>604</ns1:fid>
                    <ns1:fvalueInt>0</ns1:fvalueInt>
                </ns1:item>
            </ns1:fieldsToModify>
            <ns1:previewOnly>0</ns1:previewOnly>
        </ns1:DoChangeItemFieldsRequest>
    </SOAP-ENV:Body>
</SOAP-ENV:Envelope>`
